### PR TITLE
Add macOS builds

### PR DESCRIPTION
**Pull request by bkpoon**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

@jaimergp I started with your first commit from #8

Maybe version 2.5.0 made things simpler for building on macOS, but the main change is bumping the macOS SDK to 10.12 since that is the minimum supported version for version 2.4+ (https://pymol.org/2/support.html?#installation)

I also had to increase the tolerance for the test since it was failing on linux.

And I tested the Python 3.9 package from the Azure artifact ([uploaded to `cctbx-dev` channel](https://anaconda.org/cctbx-dev/pymol-open-source)) by loading up a simple model.

Let me know if anything needs to be changed. I want to folllow up with the `osx-arm64` build and a Python 3.10 build (`pmw` migration has not been merged).

Thanks!

### CODE REVIEW

**Review by padix-key — COMMENTED** [review]
Thank you for the effort. I am glad, that the recipe finally works for MacOS. I have a few questions regarding the changes though.

File: recipe/meta.yaml
Comment (bkpoon) [reply recipe/meta.yaml:29]: For `numpy`, there is a global pinning for that package for each Python version in https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml. I remember the pinning used to follow 1.16 for Python versions earlier than 3.9 and 1.19 for 3.9 and later, but it looks like it is 1.19 for all versions of Python now.
So yes, the global pinnings mean that this package will only be compatible with `numpy >= 1.19`. If there is some package that requires an older version of `numpy`, that could be an issue. However, new builds of other packages should also follow the same pinning, so hopefully there are not many packages that only work with earlier versions of `numpy`.
Are there legacy plugins to PyMOL that require an earlier version of `numpy`? If so, I can add back the earlier pinning.

File: recipe/meta.yaml
Comment (bkpoon) [reply recipe/meta.yaml:37]: This is similar to the `numpy` pinning. There is a global pin that is currently set to 4.8.1
```
libnetcdf:
  - 4.8.1
```
in https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml. If 4.7.4 is compatible with 4.8.1, then I can add back the earlier version. If not, then there might be conflicts if someone installs another package with a `libnetcdf` dependency that is built against 4.8.1.
Generally, I would follow the global pinnings since that would help avoid compatibility issues with other packages from conda-forge.

**Comment by bkpoon** [comment]
Another note, the deployment target may need to be updated to 10.12 since the source would not compile with the 10.9 SDK (https://conda-forge.org/docs/maintainer/knowledge_base.html#requiring-newer-macos-sdks). I do not have a machine running any macOS earlier than 10.14, so I was not able to check. I can update the deployment target here or wait to see if there are users with issues.

File: recipe/meta.yaml
Comment (padix-key) [reply recipe/meta.yaml:29]: I do not know any plugins that require an earlier version, but I am not familiar with a lot of PyMOL plugins. Since NumPy uses semantic versioning, I do not think that there are incompatibilities.

I suspected that without the pinning only the most recent NumPy version is used, but I was unaware of the global pinning, so this change is fine for me.

File: recipe/meta.yaml
Comment (padix-key) [reply recipe/meta.yaml:37]: This sounds reasonable to me

**Review by padix-key — APPROVED** [review]

**Comment by padix-key** [comment]
I do not have a strong opinion on this. As MacOS 10.12 is already 6 years old, I would tend to also increase the target version.

**Comment by bkpoon** [comment]
I just committed the change to the macOS deployment target. Since Schrödinger lists the minimum to be 10.12, it should be safe to set the minimum here.

**Comment by bkpoon** [comment]
@conda-forge-admin please rerender

**Comment by padix-key** [comment]
Thanks again. I would merge this PR, if you are finished with the changes.

**Comment by padix-key** [comment]
Before I merge, would you like to be added to the maintainers of this feedstock?

**Comment by bkpoon** [comment]
@conda-forge-admin please rerender

**Comment by bkpoon** [comment]
Sure, I would be happy to help.

**Pull request merged**
